File: go/gin/internal/modules/auth/module.go
package auth

import (
	"tcc/internal/modules/client"
	"tcc/internal/modules/manager"

	"github.com/gin-gonic/gin"
)

func AuthModule(router *gin.Engine) {
	service := AuthService{
		repository: struct{
			client client.ClientRepository
			manager manager.ManagerRepository
			}{
			client: client.ClientRepository{},
			manager: manager.ManagerRepository{},
		},
	}
	controller := AuthController{
		service: service,
	}

	group := router.Group("/auth")
	{
		group.POST("/login/clientes", controller.LoginClient)
		group.POST("/login/gerentes", controller.LoginManager)
	}
}

File: rust/axum/src/modules/auth/module.rs
use axum::{Router, routing::post};

use super::controller::AuthController;

pub fn auth_module() -> Router {
    let router = Router::new()
        .route("/login/clientes", post(AuthController::login_client))
        .route("/login/gerentes", post(AuthController::login_manager));

    Router::new().nest("/auth", router)
}